src/main/java/messenger/ImageMessengerNotificationBuilder.java

```diff
@@ -0,0 +1,68 @@
+package messenger;
+
+import notification.Notification;
+import notification.NotificationBuilder;
+
+import java.time.LocalDateTime;
+
+/**
+ * Created by annterina on 11.11.17.
+ */
+public class ImageMessengerNotificationBuilder implements NotificationBuilder {
+
+    private String newSenderId;
+    private String newReceiverId;
+    private String newSenderName;
+    private String newReceiverName;
+    private LocalDateTime newSendDate;
+    private String imageName;
+    private long imageSize;
+    private byte[] newSendImage;
+
+    public ImageMessengerNotificationBuilder() {}
+
+    public ImageMessengerNotificationBuilder withSenderId(String newSenderId) {
+        this.newSenderId = newSenderId;
+        return this;
+    }
+
+    public ImageMessengerNotificationBuilder withReceiverId(String newReceiverId) {
+        this.newReceiverId = newReceiverId;
+        return this;
+    }
+
+    public ImageMessengerNotificationBuilder withSenderName(String newSenderName) {
+        this.newSenderName = newSenderName;
+        return this;
+    }
+
+    public ImageMessengerNotificationBuilder withReceiverName(String newReceiverName) {
+        this.newReceiverName = newReceiverName;
+        return this;
+    }
+
+    public ImageMessengerNotificationBuilder withSendDate(LocalDateTime newSendDate) {
+        this.newSendDate = newSendDate;
+        return this;
+    }
+
+    public ImageMessengerNotificationBuilder withSendImage(byte[] newSendImage) {
+        this.newSendImage = newSendImage;
+        return this;
+    }
+
+    @Override
+    public ImageMessengerNotification build() {
+        return new ImageMessengerNotification(newSenderId, newReceiverId, newSenderName, newReceiverName, newSendDate, imageName, imageSize, newSendImage);
+    }
+
+    public ImageMessengerNotificationBuilder withImageSize(Long imageSize) {
```

Comment:
This refers to the comment about storing information about the image in `ImageMessengerNotification`. I think it would be more convenient for a user to add an image to the notification using only `withImage(MultipartFile)` than three methods adding the name, the size and the actual content. Other concern is that when adding an image is split between three methods, a user can accidentally set the image size to some value that doesn't correspond to the contents send. Later it may lead to some hard to find errors when one will use that information in another part of the application.

Reply:
First of all, I am not able to send <code>MultipartFile</code> object via stomp and Spring. Do you remember our conversation about allowed data type in stomp? After changing <code>byte[]</code> to <code>MultipartFile</code> I've got this error
![screenshot from 2017-11-12 21-28-36](https://user-images.githubusercontent.com/25595674/32703186-6cc0282a-c7f2-11e7-9ebe-e6402a237790.png)

Of course, I can investigate that, but it doesn't seem like half-an-hour-task.
What is more, how to send imageSize or imageName in notification? In AngularJS script I am allowed to use fields of specific class (like <code>x.senderName</code> or <code>x.sendImage</code>) I am not sure how far I can go with that, is it possible to use <code>x.sendImage.someMethod</code> I am not sure.

Actually, I started this task with <a href="http://www.baeldung.com/spring-file-upload">this</a> tutorial, which looked pretty good with model introducing, but I wasn't able to make it work.

Reply:
Oh, ok, I didn't know that. For me the biggest concern is just the fact, that you need to add information about the image in 3 methods using builder. Then maybe let's leave those three separate fields in `ImageMessengerNotification`, but provide only one method in the `ImageMessengerNotificationBuilder`, something like the following:
```
public ImageMessengerNotificationBuilder withImage(MultipartFile image) {
    this.imageName = image.getOriginalFilename();
    this.imageSize = ...
    this.image = ...
    return this;
}
```

Reply:
Yeah, good idea!

Actually, it would be better to juggle only <code>MultipartFile</code> objects on both sides, but I would set this as another task, cause it requires deeper insight into Conversion&Serialization.